File: лаба2/Go/laba2_2zadanie.go
package main

import "fmt"

func main() {
	fmt.Print("Введите N и M: ")
	var N, M int
	fmt.Scan(&N, &M)

	total := 0         // Общее количество выпитых банок
	emptybot := 0      // Количество пустых банок
	iterations := 0    // Количество итераций
	fmt.Println("\nДано:")
	for j := 0; j < N; j++ {
		fmt.Print("☺")
	}
	fmt.Println()

	fmt.Println("\nПроцесс:")
	for N > 0 || emptybot >= M { //выпиваем
		total += N
		emptybot += N
		N = 0
		iterations++

		for i := 0; i < emptybot; i++ {
			fmt.Print("©")
		}
		fmt.Println(" (пустые банки)")

		// обмен
		if emptybot >= M {
			N = emptybot / M // получаем банки
			emptybot %= M      // остаток пустых банок
			iterations++
			for i := 0; i < N; i++ {
				fmt.Print("☺")
			}
			fmt.Println(" (полные банки после обмена)")
		}
	}

	fmt.Printf("\nВыпитые банки: %d\n", total)
	fmt.Printf("Итерации: %d\n", iterations)
}

File: лаба2/PHP/laba2_2zadanie.php
<?php

echo "Введите M и N: ";
list($N, $M) = explode(' ', trim(fgets(STDIN)));

$total = 0; // Общее количество выпитых банок
$emptybot = 0; // Количество пустых банок
$iterations = 0; // Количество итераций

echo "\nДано:\n";
for ($j = 0; $j < $N; $j++) echo "☺";

echo "\nПроцесс:\n";
while ($N > 0 || $emptybot >= $M) { // выпиваем
    $total += $N;
    $emptybot += $N;
    $N = 0;
    $iterations++;

    for ($i = 0; $i < $emptybot; $i++) echo "©";
    echo " (пустые банки)\n";

    // обмен
    if ($emptybot >= $M) {
        $N = (int) ($emptybot / $M); // получаем банки
        $emptybot %= $M; // остаток пустых банок
        $iterations++;
        for ($i = 0; $i < $N; $i++) echo "☺";
        echo " (полные банки после обмена)\n";
    }
}

echo "\nВыпитые банки: $total\n";
echo "Итерации: " . ($iterations) . "\n"; 

?>

